### PR TITLE
Remote coverity docker

### PR DESCRIPTION
**Pull request by TimoSairiala**

### CODE REVIEW

File: coverity.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+capture:
+  build:
+    build-command: SKIP_BUILD_UNDERLAY_STEPS=true /packaging/build.sh
+
+commit:
+  connect:
+    auth-key-file: /auth_key.txt
+    stream: octomap_server2_github_ci
+    url: https://coverity.ssrc.fi:443/
+    on-new-cert: trust
```
Comment (joonas-fi) [inline coverity.yaml:10]: This file looks like very few details need to change per project: mainly the `stream` needs to be:
```
  stream: <repo name>_github_ci
```

A thought from me is that could there be a `coverity.yaml.template` in the container image and it gets in the `entrypoint-coverity.sh`:

- copy `coverity.yaml.template` to `coverity.yaml`
- replace `${repo_name}` from `coverity.yaml` with [GITHUB_REPOSITORY](https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables) (NOTE: probably the slash in `owner/reponame` needs to be transformed)

**Review by joonas-fi — CHANGES_REQUESTED** [review]

File: coverity.yaml (hunk 1 above)
Comment (TimoSairiala) [reply coverity.yaml:10]: Currently there is only that stream name which changes, but I am not sure if that is the case always. Maybe I am wiser after few more projects being scanned, but there are flags for different compilers and possible project filtering rules as well. Of course we could have that sort of generic base version and overwrite it if needed

File: entrypoint-coverity.sh [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash -eu
+
+echo "running coverity scan"
+export PATH=$PATH:/cov/bin/
+coverity scan --exclude-language java
+coverity list
```
[the remark this answers is not in the document]
Comment (TimoSairiala) [reply entrypoint-coverity.sh:6]: :+1:

File: .github/workflows/tii-octomap-server2-coverity.yaml [hunk 3]
```diff
@@ -3,20 +3,20 @@ name: tii-octomap-server2-coverity
...
           username: ${{ github.actor }}
           password: ${{ secrets.GITHUB_TOKEN }}
-      - name: Run coverity scan
-        run:
-          docker run --rm --entrypoint=/entrypoint-coverity.sh -e PROJECT_NAME=${{ github.event.repository.name }} -v "$(pwd):/main_ws/src"  ghcr.io/tiiuae/fog-ros-baseimage-builder-coverity:main
+          registry: ghcr.io
+          image: ghcr.io/tiiuae/fog-ros-baseimage-builder-coverity:main
+          options: -e PROJECT_NAME=${{ github.event.repository.name }} -v ${{ github.workspace }}:/main_ws/src
+          run: |
+            /entrypoint-coverity.sh
```
Comment (joonas-fi) [inline .github/workflows/tii-octomap-server2-coverity.yaml:22]: I would define the entrypoint in the base image (`ENTRYPOINT ["/entrypoint-coverity.sh"]`), so it doesn't need boilerplate in all call sites.

It's a small thing but this implementation will be copy-pasted to very many places.

**Review by joonas-fi — APPROVED** [review]
It looks to me about perfect now :clap: 

Unless changes are needed regarding the root-owned files discussion, this is ready for merge (maybe squash is a good choice here)

**Pull request merged**